### PR TITLE
Deque

### PR DESCRIPTION
**Pull request by satyampgt4**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

It is an implementation of Deque using doubly linked list in c++.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

 #1602

### CODE REVIEW

File: C-Plus-Plus/ds/Deque.cpp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+    q.enqueue_front(25);
+    // q will be : 25 15 5 6
+    q.enqueue_back(30);
+    // q will be : 25 15 5 6 30
+    q.enqueue_back(11);
+    // q will be : 25 15 5 6 30 11
+    q.enqueue_front(35);
+    // q will be : 35 25 15 5 6 30 11
+    q.display();
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Deque.cpp:174]: Use Dynamic Input instead of a Driver Code.

File: C-Plus-Plus/ds/Deque.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
+/************************************************************************
+
+In this Program we will implement Double Ended Queue (de_que)
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Deque.cpp]: Can you add a beginner pitch about Double Ended Queue?

File: C-Plus-Plus/ds/Deque.cpp [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+3. dequeue_back()
+4. dequeue_front()
+5. is_empty()
+6. display()
+7. display_reverse()
+8. size_deque()
+
+
+
+
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Deque.cpp]: Remove unnecessary lines

File: C-Plus-Plus/ds/Deque.cpp [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    node *head;
+    node *tail;
+    int n;
+public:
+   // constructor of tree
+    de_que()
+    {
+        head = NULL;
+        tail = NULL;
+        n=0;
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Deque.cpp]: The Formatting is inconsistent. Try to keep it consistent for better uniformity.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/ds/Deque.cpp (hunk 1 above)
Comment (satyampgt4) [reply C-Plus-Plus/ds/Deque.cpp:174]: i use the drive code to show how these function will execute

Comment (HarshCasper) [reply C-Plus-Plus/ds/Deque.cpp:174]: We need the Driver Code to help the User use the Code. Not to show how the Code is being executed. If not changed, we will not be able to merge this.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Khimit** [comment]
@HarshCasper  can you please assign this to me?? from DWoC

**Comment by HarshCasper** [comment]
> @HarshCasper can you please assign this to me?? from DWoC

PRs are not meant to be assigned.

**Comment by HarshCasper** [comment]
I'm closing the PR due to no activity. If you would like to reopen this PR to be reviewed please join the [Slack](https://join.slack.com/t/tesseractcodi-bxq5968/shared_invite/zt-ju6c0nqq-KgkTvFyxjNVbDRBIjADanw) and drop me a message.

**Pull request closed without merging**